File: fms_wails/internal/model/template.go
package model

// Template은 방화벽 규칙 템플릿입니다.
type Template struct {
	Version  string `json:"version"`
	Contents string `json:"contents"`
}

// Clone은 템플릿의 복사본을 반환합니다.
func (t *Template) Clone() *Template {
	return &Template{
		Version:  t.Version,
		Contents: t.Contents,
	}
}

File: fms_flutter/lib/models/template.dart
/// 방화벽 규칙 템플릿 모델
class Template {
  final String version;
  final String contents;

  Template({
    required this.version,
    required this.contents,
  });

  factory Template.fromJson(Map<String, dynamic> json) {
    return Template(
      version: json['version'] as String? ?? '',
      contents: json['contents'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'version': version,
      'contents': contents,
    };
  }

  Template copyWith({
    String? version,
    String? contents,
  }) {
    return Template(
      version: version ?? this.version,
      contents: contents ?? this.contents,
    );
  }
}
